ui/app/components/job-editor.js

```diff
@@ -12,12 +12,39 @@ export default class JobEditor extends Component {
 
   @tracked error = null;
   @tracked planOutput = null;
+  @tracked view = 'job-spec';
+  @tracked isEditing;
+
+  constructor() {
+    super(...arguments);
+    this.isEditing = !!(this.args.context === 'new');
+  }
+
+  toggleEdit(bool) {
+    this.isEditing = bool || !this.isEditing;
+  }
+
+  @action
+  edit() {
+    this.args.job.set(
+      '_newDefinition',
+      JSON.stringify(this.args.definition, null, 2)
+    );
+    this.toggleEdit(true);
+  }
```

Comment:
Request: could you have the default/smallcluster mirage case include a job with an HCL definition? As it stands I just get `undefined` for all cases right now and it's hard to evaluate if _newDefinition is being set as intended.

Reply:
We have test coverage for this via the `job-editor` integration test and in the `job-definition` acceptance test suite.

Touching Mirage deserves its own PR as this PR is big enough for just separate state and presentation logic. 

I originally highlighted this as adjacent work in the [umbrella ticket](https://github.com/hashicorp/nomad/issues/15982) because I don't think it matters. When this goes into production, users will have a mix of jobs that have specifications and jobs that do not.

The most appropriate thing to do to account for this unforeseen UX would be [conditionally rendering](https://github.com/hashicorp/nomad/pull/16279/files#r1122238137) the toggle which is something we will need to do in production.